activation/nipost.go

```diff
@@ -414,24 +429,26 @@ func (nb *NIPostBuilder) submitPoetChallenges(
 	now := time.Now()
 
 	if curPoetRoundStartDeadline.Before(now) {
-		if len(existingRegistrations) == 0 {
-			if len(registrations) == 0 {
-				// no existing registration at all, drop current registration challenge
-				err = ErrATXChallengeExpired
-			} else {
-				// no existing registration for given poets set
-				err = ErrNoRegistrationForGivenPoetFound
-				nb.logger.Warn("revert poet configuration to previous is needed immediately",
-					zap.Error(err), log.ZShortStringer("smesherID", nodeID))
-			}
+		switch {
+		case len(existingRegistrations) == 0 && len(registrations) == 0:
+			// no existing registration at all, drop current registration challenge
 			return nil, fmt.Errorf(
 				"%w: poet round has already started at %s (now: %s)",
-				err,
+				ErrATXChallengeExpired,
 				curPoetRoundStartDeadline,
 				now,
 			)
+		case len(existingRegistrations) == 0:
+			// no existing registration for given poets set
+			nb.logger.Panic(
```

Comment:
NIT: Fatal will quit w/o printing a nasty call stack, which is not needed here.
```suggestion
			nb.logger.Fatal(
```

Reply:
Using log.Fatal() is not recommended in case, if you want to test this branch of code execution
https://stackoverflow.com/questions/30688554/how-to-test-go-function-containing-log-fatal